### PR TITLE
magento/magento2#12396: Total Amount cart rule without tax

### PR DESCRIPTION
**Pull request by AleksLi**
Added new condition type to give user opportunity to choose the configuration.

### Description (*)
I thought about the fix of what should be fixed in here. But I guess it's up to the user what to choose and what conditions to have to apply the rule. 
I decided to add new Cart Attribute to **Subtotal (Excl. Tax)** which is basically one of the totals _base_subtotal_with_discount_ 

### Fixed Issues (if relevant)

1. magento/magento2#<[12396](https://github.com/magento/magento2/issues/12396)>: "Total Amount" cart rule without tax

### Manual testing scenarios (*)

1. Go to the Admin Panel. Open Marketing->Cart Price Rules
2. Add new rule or edit existing one
3. Under the Condition tab you will be able to see new Configuration Subtotal (Excl. Tax).

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @AleksLi. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/SalesRule/Model/Rule/Condition/Address.php
Comment (aleron75) [inline app/code/Magento/SalesRule/Model/Rule/Condition/Address.php:64]: Are you sure that the `base_subtotal_with_discount` is the Subtotal Excluding Tax?
Its name (and usage in the code) makes me think that it is not.
Maybe I am wrong, can you help me understand?
Thanks!

Comment (AleksLi) [reply app/code/Magento/SalesRule/Model/Rule/Condition/Address.php:64]: I'm pretty sure that that's the correct subtotal to apply a coupon. I've taken it from the [collect](https://monosnap.com/file/69g8LCXETuoMUYCik9zTvmec4XBTQq ) method which is always running. All that data in any case will be in the Quote.

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-4406](https://jira.corp.magento.com/browse/ENGCOM-4406) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@AleksLi thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by AleksLi** [comment]
@magento-engcom-team, @aleron75 . I can't merge the changes, please let me know how to do that, or who is willing to do that.

**Comment by aleron75** [comment]
> @magento-engcom-team, @aleron75 . I can't merge the changes, please let me know how to do that, or who is willing to do that.

Hello, you can't merge your own PR, it will be merged by the repo maintainers.
Thanks.

**Comment by AleksLi** [comment]
Thanks @aleron75 . Now, I will be aware of that.

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Comment by koenner01** [comment]
Why was this ever approved as this was clearly never properly tested...

Both "base_subtotal" and "base_subtotal_with_discount" are ALWAYS **excluding tax**.
The only difference is with "base_subtotal_with_discount" additional discounts are also in the subtotal.

The **only** real fix for this problem is adding "base_subtotal_total_incl_tax" as this is correct value.

**Comment by hostep** [comment]
@koenner01: I can see in our preview access to Magento 2.3.7 that it will be fixed there (and was already fixed before in 2.4.2), just FYI. That info was also provided in [this comment](https://github.com/magento/magento2/issues/29740#issuecomment-726946409).

**Pull request merged**
